pkg/config/loader.go

```diff
@@ -105,36 +125,74 @@ func ParseWinPDTargets(log logr.Logger, vpcCniConfigMap *v1.ConfigMap) (warmIPTa
 		warmPrefixTargetStr, foundWarmPrefix = vpcCniConfigMap.Data[WinWarmPrefixTarget]
 	}
 
-	// If no configuration is found, return 0
-	if !foundWarmIP && !foundMinIP && !foundWarmPrefix {
-		return warmIPTarget, minIPTarget, warmPrefixTarget
-	}
-
+	// If warm IP target config value is not found, or there is an error parsing it, the value will be set to zero
 	if foundWarmIP {
 		warmIPTargetInt, err := strconv.Atoi(warmIPTargetStr)
 		if err != nil {
-			log.Error(err, "failed to parse warm ip target", "warm ip target", warmIPTargetStr)
+			log.V(1).Info("could not parse warm ip target, defaulting to zero", "warm ip target", warmIPTargetStr)
+			warmIPTarget = 0
 		} else {
 			warmIPTarget = warmIPTargetInt
+
+			// Handle secondary IP mode scenario where WarmIPTarget is explicitly configured to zero
+			// In such a case there must always be 1 warm IP to ensure that the warmpool is never empty
+			if !isPDEnabled && warmIPTarget == 0 {
+				log.V(1).Info("Explicitly setting WarmIPTarget zero value not supported in secondary IP mode, will override with 1")
+				warmIPTarget = 1
+			}
 		}
+	} else {
+		log.V(1).Info("could not find warm ip target in ConfigMap, defaulting to zero")
+		warmIPTarget = 0
```

Comment:
Is `warmIPTarget` a passed in argument? Why do we change it to `0` here?

Reply:
Good question as the behaviour intended is something I would want your feedback on. The workflow will be as follows which is why zero is set where you highlighted.

1. Retrieve the configuration parameters from the ConfigMap.
1. Check if config map does exists. If exists proceed otherwise return default values
1. Check if all three config parameter values (warm-ip-target, min-ip-target, warm-prefix-target) are 0 or unset. 
   1. If some values set, then only use the set values and set all other values to zero hence why this statement sets the value to zero. This ensures that the behaviour for prefix delegation mode is preserved as initially intended as specified in https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/master/docs/windows/prefix_delegation_config_options.md. See thread where @jiechen0826 recommended this as discussed in https://github.com/aws/amazon-vpc-resource-controller-k8s/pull/443#discussion_r1683360844. 
   1. If all values are set to zero then use default values
1. Return the results.

Reply:
@haouc I am resolving the conversation to unblock the PR. Let me know if any concerns.